monai/transforms/utils.py

```diff
@@ -2051,5 +2058,46 @@ def has_status_keys(data: torch.Tensor, status_key: Any, default_message: str =
     return True, None
 
 
+def distance_transform_edt(
+    img: NdarrayOrTensor, sampling: None | float | list[float] = None, force_scipy: bool = False
+) -> NdarrayOrTensor:
+    """
+    Euclidean distance transform, either GPU based with CuPy / cuCIM
+    or CPU based with scipy.ndimage.
+    Choice depends on cuCIM being available or scipy can be forced with the ``force_scipy`` flag.
+
+    Note that the runtime running on the CPU may be really depending on the inputs size.
+
+    Args:
+        img: Input image on which the distance transform shall be run.
+            For CuPy shape must be ([channel, ]spatial_dim1, spatial_dim2 [, spatial_dim3]), so 2D, 3D and 4D input only.
+            If you need to run the transform on other shapes, use the ``force_scipy`` flag.
+            4D input gets passed channel-wise to cupy.
+        sampling: Spacing of elements along each dimension. If a sequence, must be of length equal to the input rank;
+            if a single number, this is used for all axes. If not specified, a grid spacing of unity is implied.
+        force_scipy: Force the CPU based scipy implementation of the euclidean distance transform
+    """
+    distance_transform_edt, has_cucim = optional_import(
+        "cucim.core.operations.morphology", name="distance_transform_edt"
+    )
+
+    if has_cp and has_cucim and not force_scipy:
+        img_ = convert_to_cupy(img)
+        if img_.ndim == 4:
```

Comment:
Thanks for the PR, I would suggest all the input should be channel-first since most of the monai transform follows.
https://github.com/Project-MONAI/MONAI/blob/e5f933781b1a5f4eb7d278e06389d531bff38dca/monai/transforms/utils.py#L969
Thanks!

Reply:
So if I understand it correctly, you propose to always assume the first channel is the dimension. That means we `assert img_.ndim > 1` and then always apply transform channel-wise, even for scipy?
Thanks!

```
out = []
            for channel in img_:
                out.append(distance_transform_edt(channel, sampling=sampling))
            distance = cp.stack(out)
```

Reply:
How about we use the spatial_dims=2/3 as a first mandatory parameter like we do here: https://docs.monai.io/en/stable/transforms.html#probnms Right now we can still change that interface the way we want. This way we can have an arbitrary number of extra dimensions in front

Reply:
Sorry, my bad. I checked the code again. I think we can just keep spatial dim here.
https://github.com/Project-MONAI/MONAI/blob/e5f933781b1a5f4eb7d278e06389d531bff38dca/monai/metrics/surface_distance.py#L181
https://github.com/Project-MONAI/MONAI/blob/e5f933781b1a5f4eb7d278e06389d531bff38dca/monai/metrics/hausdorff_distance.py#L188

Reply:
But since it's such a common pattern, should we add support for non_spatial dims as well? Then not every user of this function needs to implement this pattern.